pkg/resources/management.cattle.io/v3/globalrole/validator.go

```diff
@@ -121,49 +125,50 @@ func (a *admitter) Admit(request *admission.Request) (*admissionv1.AdmissionResp
 		return nil, err
 	}
 
+	// check if the user has the escalate verb
+	hasEscalate, escErr := auth.RequestUserHasVerb(request, gvr, a.sar, escalateVerb, newGR.Name, "")
+	if escErr != nil {
+		logrus.Warnf("Failed to check for the 'escalate' verb on GlobalRoles: %v", escErr)
+		hasEscalate = false
+	}
+
+	// if we have the escalate verb, no need to check global/cluster/namespaced permissions
+	if hasEscalate {
+		return admission.ResponseAllowed(), nil
+	}
+
 	// check for escalation separately between cluster permissions and global permissions to prevent crossover
 	clusterRules, err := a.grbResolver.GlobalRoleResolver.ClusterRulesFromRole(newGR)
 	if err != nil {
 		return nil, fmt.Errorf("unable to resolve rules for new global role: %w", err)
 	}
 
-	hasEscalate, err := a.isRulesAllowed(request, clusterRules, newGR.Name, a.grbResolver)
+	// collect all escalation errors for the user
+	var allErrors []error
+	err = auth.ConfirmNoEscalation(request, clusterRules, "", a.grbResolver)
 	if err != nil {
-		return admission.ResponseFailedEscalation(err.Error()), nil
-	}
-	if hasEscalate {
-		// if we have the escalate verb, no need to check global permissions
-		return admission.ResponseAllowed(), nil
+		allErrors = append(allErrors, err)
 	}
 
-	rules := a.grbResolver.GlobalRoleResolver.GlobalRulesFromRole(newGR)
-	// don't need to check if this request was allowed due to escalation since this is the last permission check
-	// if others are added in the future a short-circuit here will be needed like for the clusterRules
-	_, err = a.isRulesAllowed(request, rules, newGR.Name, a.resolver)
+	globalRules := a.grbResolver.GlobalRoleResolver.GlobalRulesFromRole(newGR)
+	err = validateRules(globalRules)
+	if err != nil {
```

Comment:
I would recommend returning here rather than continuing on. I'm not sure how the SAR check will react to invalid rules being used in an escalation check - better to force the user to supply valid rules first IMO.